src/core/service.c

```diff
@@ -2441,6 +2490,109 @@ static int service_deserialize_item(Unit *u, const char *key, const char *value,
                         s->watchdog_override_enable = true;
                         s->watchdog_override_usec = watchdog_override_usec;
                 }
+        } else if (streq(key, "main-command") || streq(key, "control-command")) {
+                bool control, found = false;
+                ServiceExecCommand id;
+                ExecCommand *command = NULL;
+                const char *word, *parser_state;
+                _cleanup_free_ char *args = NULL, *path = NULL;
+                _cleanup_strv_free_ char **argv = NULL;
+                size_t l;
+                int index, i;
+
+                enum ExecCommandState {
+                        STATE_EXEC_COMMAND_TYPE,
+                        STATE_EXEC_COMMAND_ID,
+                        STATE_EXEC_COMMAND_PATH,
+                        STATE_EXEC_COMMAND_ARGS,
+                        _STATE_EXEC_COMMAND_MAX,
+                        _STATE_EXEC_COMMAND_INVALID = -1,
+                } state;
+
+                control = streq(key, "control-command");
+
+                state = STATE_EXEC_COMMAND_TYPE;
+
+                FOREACH_WORD(word, l, value, parser_state) {
+                        _cleanup_free_ char *temp = NULL;
+
+                        temp = strndup(word, l);
+                        if (!temp)
+                                return log_oom();
```

Comment:
hmm, so, we can't really recover from real errors here. I think the best we can do on OOM is log, stop processing this line but proceed with the next, and hence not return with -ENOMEM here.